From a8fbad12bf7da9ddfd74ec63285e7bd76ce07843 Mon Sep 17 00:00:00 2001
From: Pekka Vuorela <pekka.vuorela@jolla.com>
Date: Wed, 22 Apr 2020 17:43:06 +0300
Subject: [PATCH] Fix Qt5 backend crashing when running in Qt application

Creating a dummy QCoreApplication instance is bad in itself,
but even worse when running in an app that already has one.
---
 libmediaart/extractqt.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/libmediaart/extractqt.cpp b/libmediaart/extractqt.cpp
index 6114270..580f3dd 100644
--- a/libmediaart/extractqt.cpp
+++ b/libmediaart/extractqt.cpp
@@ -59,6 +59,10 @@ media_art_plugin_init (gint max_width)
 	max_width_in_bytes = max_width;
 
 #ifdef HAVE_QT5
+	if (QCoreApplication::instance()) {
+		// avoid a new instance if already running inside a qt app
+		return;
+	}
 	app = new QCoreApplication (argc, argv);
 #else  /* HAVE_QT4 (we fallback to Qt4) */
  	app = new QApplication (argc, argv, QApplication::Tty);
-- 
2.25.4

